### PR TITLE
feat (MatOfPoint) test shape getter

### PR DESCRIPTION
**Pull request by giovanni-nic-kuama**
address [ADS-13] Test MatOfPoint Extension

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -10,6 +10,11 @@ android {
     compileSdkVersion 32
     buildToolsVersion "30.0.3"
 
+    testOptions {
+        unitTests.returnDefaultValues = true
+
```
Comment (matteoKuama) [inline app/build.gradle]:
```suggestion
```

File: app/build.gradle [hunk 2]
```diff
@@ -10,6 +10,11 @@ android {
     compileSdkVersion 32
     buildToolsVersion "30.0.3"
 
+    testOptions {
+        unitTests.returnDefaultValues = true
```
Comment (matteoKuama) [inline app/build.gradle]:
```suggestion
```

Comment (matteoKuama) [reply app/build.gradle]: do you need it? it changes the exceptions thrown by the test to default values (0 or null) see [https://developer.android.com/training/testing/local-tests#error](url)

File: app/build.gradle [hunk 3]
```diff
@@ -45,19 +51,19 @@ android {
 
 dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
-    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:1.7.10"
```
Comment (matteoKuama) [inline app/build.gradle]:
```suggestion
    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
leave as variable value

File: app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        )
+
+        val shape = matOfPoint.shape
+        val expected = listOf(
+            Point(0.0, 0.0),
+            Point(1.0, 0.0),
+            Point(1.0, 1.0),
+            Point(0.0, 1.0))
+
+        assertEquals(shape, expected)
```
Comment (matteoKuama) [inline app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt]: wouldn't it be better to manage the test to have a different order in the list of points? 
`val matOfPoint = MatOfPoint(
            Point(0.0, 1.0)
            Point(0.0, 0.0),
            Point(1.0, 0.0),
            Point(1.0, 1.0),
        )
val expected = listOf(
            Point(0.0, 0.0),
            Point(1.0, 0.0),
            Point(1.0, 1.0),
            Point(0.0, 1.0))` at least?

File: app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import net.kuama.documentscanner.utils.TestUtils
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.opencv.core.MatOfPoint
+import org.opencv.core.Point
+
+@RunWith(AndroidJUnit4::class)
+class MatOfPointExtensionInstrumentedTest {
```
Comment (matteoKuama) [inline app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt:13]: It seems more a unit test, does it run on an emulator?
```suggestion
class MatOfPointExtensionTest {
```

File: app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package net.kuama.documentscanner.utils
+
+import androidx.test.platform.app.InstrumentationRegistry
+
+object TestUtils {
+
```
Comment (matteoKuama) [inline app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt]:
```suggestion
```

File: app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package net.kuama.documentscanner.utils
+
+import androidx.test.platform.app.InstrumentationRegistry
+
```
Comment (matteoKuama) [inline app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt:5]:
```suggestion
import net.kuama.documentscanner.data as scanner
```

File: app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+object TestUtils {
+
+    fun loadOpenCvLibrary() {
+        System.loadLibrary("opencv_java4")
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        net.kuama.documentscanner.data.OpenCVLoader(appContext)
```
Comment (matteoKuama) [inline app/src/androidTest/java/net/kuama/documentscanner/utils/TestUtils.kt]:
```suggestion
        scanner.OpenCVLoader(appContext)
```

**Review by matteoKuama — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt (hunk 5 above)
Comment (giovanni-nic-kuama) [reply app/src/androidTest/java/net/kuama/documentscanner/extensions/MatOfPointExtensionInstrumentedTest.kt:13]: Yes. It has to run on the emulator otherwise this test cannot be successful. To load OpenCv we need the context, which is present only on instrumented test.

**Review by matteoKuama — APPROVED** [review]

**Review by matteoKuama — APPROVED** [review]
